### PR TITLE
feat(client): Add the internal behaviors API.

### PR DESCRIPTION
**Pull request by shane-tomlinson**

### CODE REVIEW

File: app/scripts/models/auth_brokers/fx-sync.js [hunk 1]
```diff
@@ -19,21 +13,38 @@ define([
...
+     * _commands must be overridden with an object that contains:
+     * {
+     *   CAN_LINK_ACCOUNT: <specify in subclass>,
+     *   CHANGE_PASSWORD: <specify in subclass>,
+     *   DELETE_ACCOUNT: <specify in subclass>,
+     *   LOADED: <specify in subclass>,
+     *   LOGIN: <specify in subclass>\
+     * }
+     */
+    _commands:null,
```
Comment (pdehaan) [inline app/scripts/models/auth_brokers/fx-sync.js]: nit: Add space after semi.
(wonders if ESLint or JSCS catches this stuff)

File: app/scripts/models/auth_brokers/fx-sync.js [hunk 2]
```diff
@@ -19,21 +13,38 @@ define([
...
+    type: 'fx-sync',
+
+    /**
+     * _commands must be overridden with an object that contains:
+     * {
+     *   CAN_LINK_ACCOUNT: <specify in subclass>,
+     *   CHANGE_PASSWORD: <specify in subclass>,
+     *   DELETE_ACCOUNT: <specify in subclass>,
+     *   LOADED: <specify in subclass>,
+     *   LOGIN: <specify in subclass>\
```
Comment (pdehaan) [inline app/scripts/models/auth_brokers/fx-sync.js]: nit: stray "\" at EOL.

File: app/scripts/models/auth_brokers/fx-sync.js (hunk 1 above)
Comment (shane-tomlinson) [inline app/scripts/models/auth_brokers/fx-sync.js]: thanks @pdehaan, I didn't notice that!

Comment (pdehaan) [inline app/scripts/models/auth_brokers/fx-sync.js]: I'm thinking we need to set the `key-spacing` rule in our .eslintrc file (or ideally upstream in the **eslint-config-fxa** module).

```js
"key-spacing": [2, { "beforeColon": false, "afterColon": true }]
```

If I try changing that locally against *master* branch, I get the following output:
```sh
➜  fxa-content-server git:(master) ✗ grunt eslint --force
Running "eslint:grunt" (eslint) task


Running "eslint:app" (eslint) task

/Users/pdehaan/dev/fxa-content-server_pd/fxa-content-server/app/scripts/views/force_auth.js
  71:17  error  Extra space before value for key "email"  key-spacing

/Users/pdehaan/dev/fxa-content-server_pd/fxa-content-server/app/tests/spec/models/auth_brokers/oauth.js
  107:26  error  Extra space before value for key "redirect"  key-spacing
  148:26  error  Extra space before value for key "redirect"  key-spacing
  169:26  error  Extra space before value for key "redirect"  key-spacing

✖ 4 problems (4 errors, 0 warnings)

Warning: Task "eslint:app" failed. Used --force, continuing.

Running "eslint:tests" (eslint) task


Running "eslint:scripts" (eslint) task


Running "eslint:server" (eslint) task


Done, but with warnings.
```

**Comment by shane-tomlinson** [comment]
@vladikoff - This is a cleaned up version of the behaviors we spoke about yesterday. Mind having a look?

**Pull request merged**